File: Day004/python/opencv/readvideo.py
import cv2

# cap=cv2.VideoCapture(0)
cap =cv2.VideoCapture("../img/1.MP4")
while True:
    ret,frame=cap.read()
    cv2.imshow('frame',frame)

    if cv2.waitKey(1)&0XFF==ord('q'):#防止有的操作系统waitkey返回值大于8位
        break

cap.release()
cv2.destroyAllWindows()

File: Day004/C++/Opencv/read_video.cpp
//
// Created by 刘 on 2022/7/21.
//视频读取

#include <opencv2/opencv.hpp>
#include <conio.h>
using namespace std;

int main(){
    cv::VideoCapture cap;
    cap=cv::VideoCapture("rtsp://wowzaec2demo.streamlock.net/vod/mp4:BigBuckBunny_115k.mp4");
    while (true){
        cv::Mat frame;
        cap>>frame;//按类型返回只能接收到图像返回值，接收不到布尔类型ret
        cv::imshow("video",frame);
        cv::waitKey(42);
        if(_kbhit()){
            break;
        }
    }
    cap.release();
    cv::destroyAllWindows();
}